django/db/models/functions/comparison.py

```diff
@@ -164,7 +164,9 @@ def as_native(self, compiler, connection, *, returning, **extra_context):
         class ArgJoiner:
             def join(self, args):
                 pairs = zip(args[::2], args[1::2], strict=True)
-                return ", ".join([" VALUE ".join(pair) for pair in pairs])
+                # 'key' is wrapped in parentheses in case it's been cast using
+                # the postgres :: shortcut syntax.
+                return ", ".join(f"({key}) VALUE {value}" for key, value in pairs)
```

Comment:
This predates this change but it seems wasteful to redefine the class on every single invocation. It looks like something that could be defined at the class scope instead.

Reply:
Right, I try to keep my actual changes to code as small as possible, as is generally preferred around here.

If you defined it in the class scope, a casual reader might think it has something to do with just the general way JsonObject works, but it really is specific to postgres, so I logically understand having it be scoped entirely within the function. 

Also, if you defined it in the class scope, it might have some sort of implication about what you're supposed to do if you were to subclass or extend JsonObject. 

I suspect the actual CPU overhead of redefining the class is pretty minimal. Would need to profile to be absolute sure of course.

Reply:
Created a PR to capture this https://github.com/django/django/pull/18768